### ISSUE TITLE
Determine if there are other methods to add to RSiteCatalyst

### ISSUE THREAD

**Issue by randyzwitch** [randyzwitch/RSiteCatalyst#179]
In writing better documentation, it appears there are more methods available in the API. See if any are worth adding.

**Comment by beskelton** [comment]
Hey Randy - big fan of the package still. Can I leave a vote for the new Segments method? No way currently to export in the regular interface. Same with Calculated metrics. Is there perhaps a way to test these methods by hacking together a request via ApiRequest and adding the Segments.Get as the func.name?

**Comment by randyzwitch** [comment]
Partial Segments example:

```
{
  "accessLevel":"all",
  
  "selected":[
    "s300001357_55f07ec7e4b0548b72256453"
    ]
}
```

```R
library(jsonlite)
request.body <- c()
request.body$accessLevel <- unbox('all')
request.body$selected <- c('s300001357_55f07ec7e4b0548b72256453')
toJSON(request.body)
```

**Comment by randyzwitch** [comment]
@beskelton How is the `Segments.Get` method different from `GetSegments()` that is already defined within RSiteCatalyst? It wouldn't be hard for me to do, but it seems like Segments.Get is just two columns, whereas `GetSegments` returns more information

**Comment by beskelton** [comment]
Hi @randyzwitch thanks for circling back, and a big thanks for your continued efforts on this package. The ReportSuites.GetSegments does work well to pull back basic segment info, however it appears to be [soon deprecated](https://marketing.adobe.com/developer/documentation/analytics-administration-1-4/r-getsegments-1). The newer Segments.Get is part of enhanced Segment capabilities of the unified Analytics Segmentation interface, and has similar methods to Save and Delete Segments as well. Segments.Get has more filtering capabilities, and you can optionally choose which fields you want listed in the response. Happy to provide sample calls and responses if that would be helpful.

**Comment by slin30** [comment]
@randyzwitch: Have been using the Segments API methods quite a bit for work recently as well. As @beskelton mentioned, `Segments.Get` does quite a bit more than `ReportSuite.GetSegments`. Am currently reusing `ApiRequest()` to craft calls to `Segments.Get` to pull/check detailed segments info.

The call is relatively straightforward; for example, draft roxygen documentation:

```{r}
#' Get Adobe Analytics Segments information
#' 
#' Query the AA segments API to return segment information based on different criteria, at 
#' the level of desired detail
#' 
#' @importFrom RSiteCatalyst ApiRequest
#' @importFrom jsonlite unbox toJSON
#'
#' @param accessLevel (optional) A character vector of length 1. Must be one of \code{all, shared, owned}. 
#' If not specified, defaults to \code{owned}. 
#' @param fields (optional) A character vector denoting the quantity, depth, and general detail of information desired. 
#' Must be one of 
#' \code{tags, shares, description, owner, modified, compatibility, favorite, reportSuiteID, definition}. 
#' The API always includes \code{id} and \code{name}, by default.
#' @param selected (optional) A character vector of segment ID(s) you wish to query for. If both \code{selected} and 
#' \code{accessLevel} are provided, \code{selected} take precedence.
#' @param sort (optional) A character vector of length 1. Must be one of \code{id, name, description, reportSuiteID,
#' owner, modified, favorite}. If not specified, defaults to \code{id}.
#' @param filters (optional) A named \code{list}. 
#' @param handle_tagsCol (optional) A logical vector of length 1
#' @param ... Additional args to pass to \code{ApiRequest}
```

Restructuring the return is another story altogether, especially for e.g. nested containers/rules and stacked segments when requesting a segment definition within the `fields` argument.

**Comment by randyzwitch** [comment]
Thanks @beskelton, deprecation is always a good reason to switch :)

@slin30, if you already have half the solution done, I'd love it if you submitted a pull request. I'm willing to review/clean it up if you need help, but having something working to start from makes it much easier for me to complete.

**Comment by slin30** [comment]
@randyzwitch: More than happy to submit a PR, and yes, will almost certainly need your input re: style, cleanup, etc. Current progress is tracked in a private company gitlab repo focused on Admin and Segments API methods-- mainly because testing the Admin methods in particular requires highly elevated access rights, and I wanted to neutralize the (admittedly remote) possibility of doing something silly like committing credentials to VC. 

Would you be opposed to reviewing current progress in a mirror within my personal (but still private) gitlab? No sensitivities around IP or anything; was considering asking about joining efforts, but never got around to addressing style/conventions-- and if you're on board, it'd provide much-needed motivation to give this project the attention it deserves (on my part).

**Comment by randyzwitch** [comment]
@slin30, yes, I would be willing to review the changes you've made and help you stage them into pull requests. I'm less worried about style; working beats style any time.

**Comment by slin30** [comment]
@randyzwitch, excellent, and thanks much. Will add you as soon as gitlab is back up. Looks like they were having a bad day that just got worse.
